lib/labels/labels.go

```diff
@@ -147,12 +149,24 @@ func (l *Dynamic) periodicUpdateLabel(name string, label types.CommandLabel) {
 
 // updateLabel will run a command, then update the value of a label.
 func (l *Dynamic) updateLabel(name string, label types.CommandLabel) {
-	out, err := exec.Command(label.GetCommand()[0], label.GetCommand()[1:]...).Output()
+	var logMsg string
+	var out bytes.Buffer
+	var stderr bytes.Buffer
+	cmd := exec.Command(label.GetCommand()[0], label.GetCommand()[1:]...)
+	cmd.Stdout = &out
+	cmd.Stderr = &stderr
+	err := cmd.Run()
 	if err != nil {
-		l.c.Log.Errorf("Failed to run command and update label: %v.", err)
-		label.SetResult(err.Error() + " output: " + string(out))
+		if logrus.GetLevel() == logrus.DebugLevel {
```

Comment:
You generally want to use `IsLevelEnabled(logrus.DebugLevel)` instead of a strict equality check, as level trace should also print debug info.

That said, I think this is too complicated. Let's just always log more information.